tree/treeplayer/test/dataframe/dataframe_simple_tests.hxx

```diff
@@ -162,6 +162,29 @@ TEST(TEST_CATEGORY, Define_jitted_complex)
    EXPECT_EQ(7.867497533559811628, *m);
 }
 
+TEST(TEST_CATEGORY, Define_jitted_complex_array_sum)
+{
+   TDataFrame tdf(10);
+   auto d = tdf.Define("x", "3.0")
+               .Define("y", "4.0")
+               .Define("z", "12.0")
+               .Define("v", "std::array<double, 3> v{x, y, z}; return v;")
+               .Define("r", "double r2 = 0.0; for (auto&& w : v) r2 += w*w; return sqrt(r2);");
+   auto m = d.Max("r");
+   EXPECT_DOUBLE_EQ(13.0, *m);
+}
+
+TEST(TEST_CATEGORY, Define_jitted_defines_with_return)
+{
+   TDataFrame tdf(10);
+   auto d = tdf.Define("my_return_x", "3.0")
+               .Define("return_y", "4.0 // with a comment")
+               .Define("v", "std::array<double, 2> v{my_return_x, return_y}; return v; // also with comment")
```

Comment:
If we can strip out comments, it's the best solution. However, if the compiler understands the comment (and it will in the example above), then there should be no problems. Only comments that extend to the end of the line are a bit problematic, but I added a newline to avoid problems there. What's the main problem that we cannot solve is a comment with "return" inside. For that to work, we need to strip out comments.

Reply:
Code manipulation such as stripping out comments can only lead to trouble, and it doesn't really add anything to the feature (users can just put their comments outside of the string).

See my comment in the main PR thread: in my opinion we can either explicitly state we do not support comments in expression strings, or not say anything and hope that users do not write `return` or branch names in the comments in the expression string.